lib/src/wallet/traits.rs

```diff
@@ -0,0 +1,464 @@
+use super::{
+    data::{OrchardNoteData, SaplingNoteData, WalletNullifier, WalletTx, WitnessCache},
+    keys::{orchard::OrchardKey, sapling::SaplingKey},
+};
+use nonempty::NonEmpty;
+use orchard::{
+    bundle::{Authorization as OrchardAuthorization, Bundle as OrchardBundle},
+    keys::{
+        Diversifier as OrchardDiversifier, FullViewingKey as OrchardFullViewingKey,
+        IncomingViewingKey as OrchardIncomingViewingKey,
+        OutgoingViewingKey as OrchardOutgoingViewingKey, SpendingKey as OrchardSpendingKey,
+    },
+    note::{Note as OrchardNote, Nullifier as OrchardNullifier},
+    note_encryption::OrchardDomain,
+    tree::MerkleHashOrchard,
+    Action, Address as OrchardAddress,
+};
+use zcash_address::unified::{self, Encoding as _, Receiver};
+use zcash_client_backend::encoding::encode_payment_address;
+use zcash_note_encryption::{Domain, ShieldedOutput, ENC_CIPHERTEXT_SIZE};
+use zcash_primitives::{
+    consensus::Parameters,
+    keys::OutgoingViewingKey as SaplingOutgoingViewingKey,
+    memo::{Memo, MemoBytes},
+    merkle_tree::Hashable,
+    sapling::{
+        note_encryption::SaplingDomain, Diversifier as SaplingDiversifier, Node as SaplingNode,
+        Note as SaplingNote, Nullifier as SaplingNullifier, PaymentAddress as SaplingAddress,
+        SaplingIvk,
+    },
+    transaction::{
+        components::{
+            sapling::{
+                Authorization as SaplingAuthorization, Bundle as SaplingBundle, GrothProofBytes,
+            },
+            OutputDescription, SpendDescription,
+        },
+        TxId,
+    },
+    zip32::{
+        ExtendedFullViewingKey as SaplingExtendedFullViewingKey,
+        ExtendedSpendingKey as SaplingExtendedSpendingKey,
+    },
+};
+use zingoconfig::Network;
+
+pub(crate) trait ToBytes<const N: usize> {
+    fn to_bytes(&self) -> [u8; N];
```

Comment:
Is this for the purpose of providing a standard interface?

Reply:
Yes.